renku/domain_model/template.py

```diff
@@ -148,40 +149,64 @@ def get_raw_content(self) -> List[Dict]:
-            if not isinstance(template, dict):
-                raise errors.InvalidTemplateError(f"Invalid template type: '{type(template).__name__}'")
+        for template_entry in self._content:
+            if not isinstance(template_entry, dict):
+                raise errors.InvalidTemplateError(f"Invalid template type: '{type(template_entry).__name__}'")
 
-            id = template.get("id") or template.get("folder")
+            id = template_entry.get("id") or template_entry.get("folder")
             if not id:
-                raise errors.InvalidTemplateError(f"Template doesn't have an id: '{template}'")
+                raise errors.InvalidTemplateError(f"Template doesn't have an id: '{template_entry}'")
+            if not template_entry.get("folder"):
+                warnings.append(f"Template '{id}' should use 'folder' attribute instead of 'id'.")
 
-            parameters = template.get("parameters") or template.get("variables")
+            parameters = template_entry.get("parameters") or template_entry.get("variables")
             if parameters:
+                if template_entry.get("parameters"):
+                    warnings.append(f"Template '{id}' should use 'variables' instead of 'parameters' in manifest.")
+
+                if not template_entry.get("variables"):
+                    template_entry["variables"] = {}
+
                 if not isinstance(parameters, dict):
-                    raise errors.InvalidTemplateError(f"Invalid template variable type: '{type(parameters).__name__}'")
+                    raise errors.InvalidTemplateError(
+                        f"Invalid template variable type on template '{id}': '{type(parameters).__name__}'"
+                    )
 
                 for key, parameter in parameters.items():
                     if isinstance(parameter, str):  # NOTE: Backwards compatibility
-                        template["variables"][key] = {"description": parameter}
+                        template_entry["variables"][key] = {"description": parameter}
+                        warnings.append(
+                            f"Template '{id}' variable '{key}' uses old string format in manifest and should be"
+                            " replaced with the nested dictionary format."
+                        )
+        if not manifest_only:
+            for template in self.templates:
+                template.validate(skip_files=True)
 
-        for template in self.templates:
-            template.validate(skip_files=True)
+        return warnings
 
 
 class Template:
     """Template files and metadata from a template source."""
 
     REQUIRED_ATTRIBUTES = ("name",)
     REQUIRED_FILES = (os.path.join(RENKU_HOME, "renku.ini"), "Dockerfile")
+    PROHIBITED_PATHS = (
+        os.path.join(RENKU_HOME, "metadata"),
+        os.path.join(RENKU_HOME, "metadata.yml"),
+        os.path.join(RENKU_HOME, "template_checksums.json"),
+        os.path.join(RENKU_HOME, "cache"),
```

Comment:
We should add `tmp`, `pointers`, `datasets`, ... here as well. Perhaps it's easier/safer to prohibit all paths in `RENKU_HOME` except `renku.ini`.